app/components/create-proposal/create-proposal-preview-step/create-proposal-preview-step.component.tsx

```diff
@@ -55,10 +59,36 @@ export const CreateProposalPreviewStep = () => {
     };
   }, [form]);
 
-  const { data, error, writeContract } = useWriteContract();
+  const {
+    error: simulatedContractError,
+    isError: isSimulatedContractError,
+    isPending: isSimulatedContractPending,
+    refetch: refetchSimulatedCall,
+  } = useSimulateContract({
+    address: contracts?.MentoGovernor.address as Address,
+    abi: GovernorABI,
+    functionName: "propose",
+
+    args: [
+      proposal.transactions.map(
+        (transaction) => transaction.address as Address,
+      ),
+      proposal.transactions.map((transaction) => BigInt(transaction.value)),
+      proposal.transactions.map((transaction) => transaction.data),
+      JSON.stringify(proposal.metadata),
+    ] as any,
+  });
+
+  useEffect(() => {
+    if (form[formStep].isOpened) {
+      refetchSimulatedCall();
+    }
+  }, [form, proposal, refetchSimulatedCall]);
 
```

Comment:
You don't have to refetch, as it does it automatically. However, you can use this param to prevent it from fetching until you get to this step 

```suggestion
    query: {
      enabled: false,
    },
  });

  // useEffect(() => {
  //   if (form[formStep].isOpened) {
  //     refetchSimulatedCall();
  //   }
  // }, [form, proposal, refetchSimulatedCall]);
```